cli/cli/cmds_00_other.go

```diff
@@ -69,8 +69,7 @@ func (c *CLI) initOtherCmds() {
 }
 
 func (c *CLI) initOtherFlags() {
-	cobra.HelpFlagShorthand = "?"
-	cobra.HelpFlagUsageFormatString = "Help for %s"
+	c.c.PersistentFlags().BoolP("help", "?", false, "help for "+c.c.Name())
```

Comment:
Hi @codenrhoden,

So I've been trying to get back to mainline Cobra for some time. One of the reasons for the fork was, as you mention, the Help flag situation. I've been aware now for a little bit that they've updated the code to enable a customizeable help flag shorthand. However, one of the outstanding issues is something it doesn't appear you've circumvented with this PR, and I'm attaching this message at the point where I see the problem.

> `"help for "+c.c.Name()`

What do you think that prints? I'm pretty sure it will emit `help for rexray`, which is not appropriate when the help text is emitted for a sub-command. That's why I introduced the `HelpFlagUsageFormatString` field to enable the replacement of a `%s` with the name of the ***active*** command.

The only way to achieve the same using mainline cobra is to define the Help flag as a local flag for *all* defined commands and set the usage text of each and every one to `help for COMMAND_NAME` where `COMMAND_NAME` is the command for which the flag is being defined.